src/libsystemd/sd-event/sd-event.c

```diff
@@ -2318,7 +2318,7 @@ static int source_dispatch(sd_event_source *s) {
 
         if (r < 0)
                 log_debug_errno(r, "Event source %s (type %s) returned error, disabling: %m",
-                                strna(s->description), event_source_type_to_string(s->type));
+                                strna(s->description), strna(event_source_type_to_string(s->type)));
```

Comment:
Hm, that seems a bit ugly. We keep s->description (if set) longer than s->type. But s->description is optional. Shouldn't instead `source_disconnect` be modified to not touch s->type, so we can log somewhat-usefully about the object?